src/types/tostring.rs

```diff
@@ -34,6 +35,9 @@ impl Type {
             }
         }
     }
+    pub fn parse<F: FromStr>(&self) -> Result<F, F::Err> {
```

Comment:
What's the point of this function? Casting? If so, I think it'd be more efficient to try on a case-by-case basis.

Reply:
I've tried doing it case by case, but the problem is that primitive types just sorta don't have an easy way to
cast it to other primitives. For example to cast f32 to int we can use `<F: From<f32>>` and `F::from(n)`. But if the F is an f32,
this wouldn't work. Any ideas?

Reply:
I know how unhelpful it is, but my instinct is to say "do it any way other than this!"

I don't think it's idiomatic of Europa nor very good as a design decision in general to allow random types to be interpreted as others simply by nature of their stringified form (intended for **debugging**) being parsable to the type. In the situations in your code where you have used this function, there is a much better way of doing it, and on which I've left more commentary in the appropriate review comment. In short, you should match the type structure instead of casting through parsing debug strings.

Reply:
Fair enough then.
Will do that if possible.